test/e2e/operator_test.go

```diff
@@ -343,9 +343,10 @@ func TestLogLevel(t *testing.T) {
 
 	// wait for KCM pods to be successfully running
 	// then check that "v=4" was added to the appropriate containers
+	var lastListErr error
 	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
-		pods, err := kubeClient.CoreV1().Pods(operatorclient.TargetNamespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "app=kube-controller-manager"})
-		require.NoError(t, err)
+		var pods *corev1.PodList
+		pods, lastListErr = kubeClient.CoreV1().Pods(operatorclient.TargetNamespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "app=kube-controller-manager"})
```

Comment:
I suppose an alternative would be to just check for `connection reset by peer` errors and fail otherwise instead

Reply:
Yup. This seems like a tolerable error especially anything within loop. You can also try increasing the retry time instead of hitting the apiserver every second.